### PR TITLE
expr/builtin_fncall: implement UnHex

### PR DESCRIPTION
**Pull request by sweetIan**
## What have you changed? (mandatory)

expr/builtin_fncall: implement string.UnHex

## What are the type of the changes? (mandatory)

- New feature (non-breaking change which adds functionality)

## How has this PR been tested? (mandatory)

unit test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

no

## Does this PR affect tidb-ansible update? (mandatory)

no

## Refer to a related PR or issue link (optional)

#3275

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -79,6 +79,7 @@ crossbeam = "0.2"
 fxhash = "0.2"
 derive_more = "0.11.0"
 num = "0.2.0"
+hex="0.3.2"
```
Comment (breeswish) [inline Cargo.toml]: Please add space like `hex = "0.3"`, and also I think it is not necessary to specify the minor version.

File: src/coprocessor/dag/expr/builtin_string.rs [hunk 2]
```diff
@@ -77,6 +78,28 @@ impl ScalarFunc {
...
+            let mut vec = vec![b'0'];
+            vec.extend_from_slice(&s);
+            vec
+        } else {
+            s.to_vec()
+        };
+        let result = Vec::from_hex(hex_string);
+        result
+            .map_err(|_e| {
+                Error::wrong_value_for_type("string", str::from_utf8(s.borrow()).unwrap(), "unhex")
```
Comment (AndreMouche) [inline src/coprocessor/dag/expr/builtin_string.rs]: It seems we should return null for the following case https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_unhex
```
mysql> SELECT UNHEX('GG');
+-------------+
| UNHEX('GG') |
+-------------+
| NULL        |
+-------------+
```

Comment (sweetIan) [reply src/coprocessor/dag/expr/builtin_string.rs]: Yes, mysql also returns a warning. So we skip the warning and just return NULL, right?

**Comment by hicqu** [comment]
LGTM.

File: src/coprocessor/dag/expr/builtin_string.rs [hunk 3]
```diff
@@ -77,6 +78,24 @@ impl ScalarFunc {
...
+
+    #[inline]
+    pub fn un_hex<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let s = try_opt!(self.children[0].eval_string(ctx, row));
+        let hex_string = if s.len() % 2 == 1 {
+            let mut vec = vec![b'0'];
```
Comment (siddontang) [inline src/coprocessor/dag/expr/builtin_string.rs]: can you add a comment here?

Rest LGTM

File: src/coprocessor/dag/expr/builtin_string.rs (hunk 2 above)
Comment (AndreMouche) [reply src/coprocessor/dag/expr/builtin_string.rs]: I think we can keep compatible with `tidb` and return a Null.

**Pull request merged**